src/main/java/com/thealgorithms/dynamicprogramming/TravellingSalesmanBitManipulation.java

```diff
@@ -0,0 +1,54 @@
+package com.thealgorithms.dynamicprogramming;
+public class TravellingSalesmanBitManipulation {
+
+    /** 
+     * @brief
+     * The below algorithm is called as Travelling Salesman DP optimisation Using Bit-manipulation.
+     * We can use this algorithm to improve the  naive algorithm’s performance.
+     * @detail
+     * Given the distance/cost(as and adjacency matrix) between each city/node to the other city/node,
+     * the problem is to find the shortest possible route that visits every city exactly once
+     * and returns to the starting point, or we can say the minimum cost of whole tour.
+
+     * Explanation
+     * INPUT -> You are given with an adjacency matrix A = {} which contains the distance between two cities/node.
+     * OUTPUT ->  Minimum cost of whole tour from starting point
+     * The Worst Case Time Complexity: O(n^2 * 2^n)
+     * Space complexity: O(n)
+       [Travelling Salesman problem Dynamic Programming optimisation using bit-masking](https://www.geeksforgeeks.org/travelling-salesman-problem-set-1/)
+
+     * @author
+     * [Utkarsh Yadav](https://github.com/Rytnix)
+
+     * @param dist is a 2D Array contains distance between 2 cities.
+     * @param setOfCities takes Integer number to represent the city/node.
+     * @param city takes an Integer number to track the current city.
+     * @param n takes an Integer no to tell how many no of
+     * @param dp is a 2D Array for memorizing the states to avoid re-computation
+     * @return min cost of travelling from start to destination and back to start position.
+     */
+    public static int travellingSalesmanBits(int[][] dist,
```

Comment:
Please create a wrapper function that creates dp argument for this function and calls it. The caller doesn't need to know about `dp`.